kubeflow/trainer/backends/localprocess/utils.py

```
@@ -0,0 +1,298 @@
+    # find os python binary to create venv
+    python_bin = shutil.which("python")
+    if not python_bin:
+        python_bin = shutil.which("python3")
+    if not python_bin:
+        raise ValueError("No python executable found")
+
+    # workout if dependencies needs to be installed
+    if isinstance(runtime.trainer, LocalRuntimeTrainer):
+        runtime_trainer: LocalRuntimeTrainer = runtime.trainer
+    else:
+        raise ValueError("Invalid Runtime Trainer type: {type(runtime.trainer)}")
```

Ruff (0.13.1)

Prefer `TypeError` exception for invalid type

(TRY004)